devito/operator/profiling.py

```diff
@@ -367,21 +376,23 @@ class AdvisorProfiler(AdvancedProfiler):
     _ext_calls = [_api_resume, _api_pause]
 
     def __init__(self, name):
-        self.path = locate_intel_advisor()
-        if self.path is None:
-            self.initialized = False
+        if self._attempted_init:
+            return
+
+        super().__init__(name)
+
+        path = get_advisor_path()
+        if path:
+            self.add_include_dir(path.joinpath('include').as_posix())
+            self.add_lib_dir(path.joinpath('lib64').as_posix())
+            self._attempted_init = True
         else:
-            super().__init__(name)
-            # Make sure future compilations will get the proper header and
-            # shared object files
-            compiler = configuration['compiler']
-            compiler.add_include_dirs(self.path.joinpath('include').as_posix())
-            compiler.add_libraries(self._default_libs)
-            libdir = self.path.joinpath('lib64').as_posix()
-            compiler.add_library_dirs(libdir)
-            compiler.add_ldflags('-Wl,-rpath,%s' % libdir)
+            self._attempted_init = False
```

Comment:
useless

Reply:
This is useful to be able to revert to `advanced` IFF `icx` is available but `advisor` not.
Check this line:
https://github.com/devitocodes/devito/pull/2533/files#diff-6ef42b440b70eb3098c97da2cf40ef16d030c6566dbdc3aeb2f87cf0313c057dR525